herddb-collections/src/main/java/herddb/collections/CollectionsManager.java

```diff
@@ -66,24 +66,28 @@
 
     private static final AtomicLong TABLE_NAME_GENERATOR = new AtomicLong();
 
-    private static final ValueSerializer DEFAULT_VALUE_SERIALIZER = new ValueSerializer() {
-        @Override
-        public void serialize(Object object, OutputStream oo) throws Exception {
-            try (ObjectOutputStream ooo = new ObjectOutputStream(oo)) {
-                ooo.writeUnshared(object);
+    private static ValueSerializer defaultValueSerializer(int expectedSize) {
+        return new ValueSerializer() {
+            @Override
+            public byte[] serialize(Object object) throws Exception {
+                try (VisibleByteArrayOutputStream oo = new VisibleByteArrayOutputStream(expectedSize);
+                        ObjectOutputStream ooo = new ObjectOutputStream(oo)) {
+                    ooo.writeUnshared(object);
+                    return oo.toByteArray();
```

Comment:
Easy improvement. ToByteArray only if expected and real size differs otherwise getBuffer whould be better due to zero copy. We could even add a method with such logic